packages/mongodb-access-adapter/datagrip-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/adapter/DataGripMongoDbDriver.kt

```diff
@@ -81,7 +84,27 @@ internal class DataGripMongoDbDriver(
 
     override suspend fun connectionString(): ConnectionString = ConnectionString(dataSource.url!!)
 
-    override suspend fun <S> explain(query: Node<S>): ExplainPlan = ExplainPlan.CollectionScan
+    override suspend fun <S> explain(query: Node<S>): ExplainPlan = withContext(Dispatchers.IO) {
```

Comment:
This connects the explain plan to the query generation. It's the work we need to do for INTELLIJ-93, but it was very convenient to work on it so we could test that the generated query was running on the index from within the IDE, and it wasn't a lot of work.